h2o-algos/src/main/java/hex/tree/DTree.java

```diff
@@ -216,12 +226,12 @@ float splat(DHistogram hs[]) {
     public DHistogram[] nextLevelHistos(DHistogram currentHistos[], int way, double splat, SharedTreeModel.SharedTreeParameters parms, Constraints cs) {
       double n = way==0 ? _n0 : _n1;
       if( n < parms._min_rows ) {
-//        Log.info("Not splitting: too few observations left: " + n);
+        if (LOG.isTraceEnabled()) LOG.trace("Not splitting: too few observations left: " + n);
         return null; // Too few elements
       }
       double se = way==0 ? _se0 : _se1;
       if( se <= 1e-30 ) {
-//        Log.info("Not splitting: pure node (perfect prediction).");
+        LOG.trace("Not splitting: pure node (perfect prediction).");
```

Comment:
You don't if here for JVM optimization but the perf impact is negligible anyway.

Reply:
the if is only needed if prepring the log message contains some work, like string concat or calling a complex toString() method